### PR TITLE
Allow configuring custom ServerHttpHeadersWriter for Kotlin DSL

### PR DESCRIPTION
**Pull request by evgeniycheban**
Closes gh-16009

### CODE REVIEW

File: config/src/test/kotlin/org/springframework/security/config/web/server/ServerHeadersDslTests.kt [hunk 1]
```diff
@@ -198,4 +199,50 @@ class ServerHeadersDslTests {
...
+
+    @Configuration
+    @EnableWebFluxSecurity
+    @EnableWebFlux
+    open class ServerHttpHeadersWriterCustomConfig {
+        @Bean
+        open fun springWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
+            return http {
+                headers {
+                    serverHttpHeadersWriter {
```
Comment (jzheaux) [inline config/src/test/kotlin/org/springframework/security/config/web/server/ServerHeadersDslTests.kt:222]: While this looks nice on the Kotlin side, it's a little clumsy on the Java side. What about this instead:

```
headers {
    writer { exchange ->
    writer { exchange ->
}
```

I think this still looks quite nice on the Kotlin side and doesn't add another DSL layer on the Java side.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @evgeniycheban! Could we avoid adding the extra DSL layer? Please see my inline comment for more details.

File: config/src/test/kotlin/org/springframework/security/config/web/server/ServerHeadersDslTests.kt (hunk 1 above)
Comment (evgeniycheban) [reply config/src/test/kotlin/org/springframework/security/config/web/server/ServerHeadersDslTests.kt]: Done

**Comment by evgeniycheban** [comment]
Hi, @jzheaux I just updated PR according to your comments.

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
Great, @evgeniycheban. We'll go ahead and merge this after we cut the `6.4.x` maintenance branch.

**Comment by evgeniycheban** [comment]
@jzheaux should I add `since 6.5` tag?

**Comment by jzheaux** [comment]
@evgeniycheban, yes please, and thank you.

**Comment by evgeniycheban** [comment]
@jzheaux I added `since 6.5` tag to `writer` function.

**Comment by jzheaux** [comment]
Thanks, @evgeniycheban, for the PR! It was nice to work with you again.

**Pull request merged**